### ISSUE TITLE
Problems with results containing inaccurate words such as "bicyclebi", "bicyclebike", "bicycle motorbike", and "the".

### ISSUE THREAD

**Issue by lanhas** [IDEA-Research/GroundingDINO#209]
When the given text_prompt is “motorbike . bicycle . the telegraph pole ”, the result will contain “bicyclebi, telegraph pole, bicyclebike， telegraph, bicycle motorbike， the". I'd like to get some accurate categories that agree with the words I'm given. I know that raising the threshold or setting a condition to filter the recognition results would solve this problem, but it would result in a lot of missed boxes. How can I solve this problem?

**Comment by lanhas** [comment]
> 

and those with #, such as '###bi'.

**Comment by Dongjie-Cheng** [comment]
Hi, I just have the same problem. Do you find a solution?

**Comment by 1benwu1** [comment]
this happens a lot... still don't know how to handle it

**Comment by ztc20011103** [comment]
Adding remove_combined = True as  a parameter in your predict function will work
boxes, logits, phrases = predict(
            model=model,
            image=image,
            caption=text_prompt,
            box_threshold=BOX_THRESHOLD,
            text_threshold=TEXT_THRESHOLD,
            remove_combined = True
        )
